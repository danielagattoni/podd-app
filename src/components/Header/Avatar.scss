@import "../../styles/config.mixins.scss";
@import "../../styles/config.variables";

.Avatar {
  width: 170px;
  height: 170px;
  background-image: url('/pie-of-the-day.svg');
  background-size: cover;
  border: .5rem solid $color-accent-secondary;
  border-radius: 50%;
  animation: showAvatar .3s ease-out;
}

@keyframes showAvatar {
  0% {
    clip-path: circle(0);
  }
  100% {
    clip-path: circle(5rem);
  }
}
