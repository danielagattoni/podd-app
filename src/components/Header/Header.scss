@import "../../styles/config.mixins.scss";
@import "../../styles/config.variables";

.Header {
  grid-row: 1;
  grid-column: 1 / -1;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 10rem;
  padding-top: $gutter-v;

  &-title {
    font-size: map-get($font-size-primary, lg);
    font-weight: bold;
    text-transform: uppercase;
    margin: $gutter-v/2 0;
  }
}
