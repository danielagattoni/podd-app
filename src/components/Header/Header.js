import React from 'react';
import { NavLink } from 'react-router-dom';
import './Header.scss';
import './Avatar.scss';

const Header = () => (
  <div className="Header">
    <NavLink exact to="/"><div className="Avatar"></div></NavLink>
    <div className="Header-title">PIE OF THE DAY</div>
  </div>
);

export default Header;
