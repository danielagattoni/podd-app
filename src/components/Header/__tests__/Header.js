import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Header from 'components/Header/Header.js';

const renderer = new ShallowRenderer();

describe('Header component', () => {

  it('displays Header component', () => {
    const component = renderer.render(<Header />);
    expect(component).toMatchSnapshot();
  });
});
