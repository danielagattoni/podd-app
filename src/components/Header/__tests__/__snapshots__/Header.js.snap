// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component displays Header component 1`] = `
<div
  className="Header"
>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    <div
      className="Avatar"
    />
  </NavLink>
  <div
    className="Header-title"
  >
    PIE OF THE DAY
  </div>
</div>
`;
